cloudinit/config/cc_ubuntu_advantage.py

```diff
@@ -172,102 +181,103 @@ def supplemental_schema_validation(ua_config: dict):
     This function supplements flexible jsonschema validation with specific
     value checks to aid in triage of invalid user-provided configuration.
 
+    Note: It does not log/raise config values as they could be urls containing
+    sensitive auth info.
+
     @param ua_config: Dictionary of config value under 'ubuntu_advantage'.
 
     @raises: ValueError describing invalid values provided.
     """
     errors = []
-    nl = "\n"
     for key, value in sorted(ua_config.items()):
-        if key in (
-            "http_proxy",
-            "https_proxy",
-            "global_apt_http_proxy",
-            "global_apt_https_proxy",
-            "ua_apt_http_proxy",
-            "ua_apt_https_proxy",
-        ):
-            try:
-                parsed_url = urlparse(value)
-                if parsed_url.scheme not in ("http", "https"):
-                    errors.append(
-                        f"Expected URL scheme http/https for ua:config:{key}."
-                        f" Found: {value}"
-                    )
-            except (AttributeError, ValueError):
+        if key not in UA_CONFIG_PROPS:
+            LOG.warning(
+                "Ignoring unknown ubuntu_advantage.config.%s property", key
+            )
+            continue
```

Comment:
While this is a nice sentiment, this now tightly couples `ubuntu_advantage: config` to whatever cloud-init releases. UA currently has more config options beyond the *proxy variables, such as update_status_timer, update_messaging_timer and metering_timer. I think we still want to allow setting those values, but maybe we just log  a warning and say, "Attempting to set unknown ua config option X".  If we are too strict here, customers which may want to immediately test ua config options that cloud-init doesn't know about will get blocked and have to wait for a cloud-init SRU to support such config settings.   And our worst-case scenario is that we can provide warnings for keys which may not be supported by the underlying ubuntu-advantage-tools version, and the ua config set <invalid_name> <value> will result in a traceback, which we want to fail cloud-init in error state if unsupported.

Reply:
done